lib/Cake/Model/Datasource/Database/Sqlite.php

```diff
@@ -63,13 +63,17 @@ class Sqlite extends DboSource {
 /**
  * SQLite3 column definition
  *
+ * @link https://www.sqlite.org/datatype3.html Datatypes In SQLite Version 3
+ *
  * @var array
  */
 	public $columns = array(
 		'primary_key' => array('name' => 'integer primary key autoincrement'),
 		'string' => array('name' => 'varchar', 'limit' => '255'),
 		'text' => array('name' => 'text'),
 		'integer' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'smallint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'tinyint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
```

Comment:
Probably `ModelTask` must be updated, as it have to bake numeric validation rules for new smallint/tinyint fields. Regarding the two fixtures, I think we need not to touch it, as they would be fixtures for some test cases of `decimal` fields. Instead, we may have to create `SmallintFixture` and `TinyintFixture` to add new test cases for the new field types.